src/Pact/Runtime/Capabilities.hs

```diff
@@ -241,19 +242,21 @@ checkManaged i (applyF,installF) cap@SigCapability{} cdef = case _dDefMeta cdef
       Just (argName,_) -> view _2 <$> defCapMetaParts c argName cdef
     getStatic DefcapEvent c = return c
 
-    -- check sig and autonomous caps for match
-    -- to install.
-    checkSigs dcm = case getStatic dcm cap of
+    -- check sig, verifier and autonomous caps for match to install.
+    checkMsgCaps dcm = case getStatic dcm cap of
       Left e -> evalError' cdef e
       Right capStatic -> do
         autos <- use $ evalCapabilities . capAutonomous
-        sigCaps <- (S.union autos . S.unions) <$> view eeMsgSigs
-        foldM (matchSig dcm capStatic) Nothing sigCaps
-
-    matchSig _ _ r@Just{} _ = return r
-    matchSig dcm capStatic Nothing sigCap = case getStatic dcm sigCap of
+        sigCaps <- S.unions <$> view eeMsgSigs
+        verifierCaps <- S.unions <$> view eeMsgVerifiers
```

Comment:
aren't these both already unioned at this point since they're scon'd into the env?

Reply:
Not sure what you mean here, but eeMsgSigs doesn't include verifier caps.